packages/material-ui/src/Global/Global.test.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { expect } from 'chai';
+import { createClientRender } from 'test/utils';
+import ThemeProvider from '../styles/ThemeProvider';
+import createMuiTheme from '../styles/createMuiTheme';
+import Global from './Global';
+
+const customTheme = createMuiTheme({
+  spacing: 10,
+});
+
+describe('Global', () => {
+  const render = createClientRender();
+
+  it('should provide default theme', function () {
+    if (/jsdom/.test(window.navigator.userAgent)) this.skip();
+
+    const { container } = render(
+      <div>
+        <Global styles={(theme) => `span { margin-top: ${theme.spacing(1)}; }`} />
+        <span>Text</span>
+      </div>,
+    );
+
+    expect(container.getElementsByTagName('span')[0]).toHaveComputedStyle({
+      marginTop: '8px',
+    });
+  });
+
+  it('should respect context theme if available', function () {
+    if (/jsdom/.test(window.navigator.userAgent)) this.skip();
+
+    const { container } = render(
+      <ThemeProvider theme={customTheme}>
+        <Global styles={(theme) => `span { margin-top: ${theme.spacing(1)}; }`} />
+        <span>Text</span>
+      </ThemeProvider>,
+    );
+
+    expect(container.getElementsByTagName('span')[0]).toHaveComputedStyle({
```

Comment:
@eps1lon Should we aim for `document` over `container` anytime possible? I have been wondering for some time. Is there a recommended approach? I like how using `document` is simpler.

Reply:
I've been thinking about not exposing the `container` at all. But there are more fundamental issues with the testing approach like still insisting to test implementation details. These are more pressing issues than whether to query `container` or `document`.